crates/gitbutler-project-store/src/migrations/migrator.rs

```diff
@@ -0,0 +1,204 @@
+use anyhow::{Context, Ok, Result};
+use rusqlite::{Connection, TransactionBehavior};
+use uuid::Uuid;
+
+use super::migration::Migration;
+
+/// SQL required to create the migrations table.
+///
+/// This should only ever be changed with great caution
+const CREATE_MIGRATIONS_TABLE: &str = "
+CREATE TABLE IF NOT EXISTS migrations (
+    id BLOB PRIMARY KEY NOT NULL, -- A V4 UUID
+    name TEXT NOT NULL
+)
+";
+
+/// Migrator is used to perform migrations on a particular database.
+pub(crate) struct Migrator<'l> {
```

Comment:
When there is no name used for the lifetime then it's traditionally `'a`, which is better for readability and sticking to conventions. Then I for one don't have to wonder if `l`  means `link`, but be upset because I will never know for sure 😁.